### ISSUE TITLE
Get/Set Functions: Return NULL value?

### ISSUE THREAD

**Issue by andsv2** [statnet/EpiModel#380]
Currently if a parameter or attribute isn't found in control, init, param, etc., the get and set helper functions return an error with no output. However, in the EpiModel workflow, there are logic checks on whether a sublist is empty or not: for example, in a two-group model, if the second group infection parameter is missing, EpiModel will update it to the first group infection parameter; or (presently), if status.vector is missing on the init sublist, then init_status will randomly assign infection status at the outset.

Is it possible for the error to be pulled down to a non-executable breaking warning (output once in a simulation)? Example below:

``` n <- 100
  nw <- network.initialize(n = n, directed = FALSE)
  formation <- ~edges
  target.stats <- 50
  coef.diss <- dissolution_coefs(dissolution = ~offset(edges), duration = 20)
  est1 <- netest(nw, formation, target.stats, coef.diss, verbose = FALSE)

  # Epidemic model
  param <- param.net(inf.prob = 0.3, rec.rate = 0.01)

  status <- sample(c("s", "i"), size = n, replace = TRUE, prob = c(0.8, 0.2))
  infTime <- rep(NA, n)
  infTime[which(status == "i")] <- -rgeom(sum(status == "i"), prob = 0.01) + 2
  init <- init.net(status.vector = status, infTime.vector = infTime)

  control <- control.net(type = "SIS", nsteps = 100, nsims = 5, verbose.int = 0)
  mod1 <- netsim(est1, param, init, control)```

**Comment by AdrienLeGuillou** [comment]
Several workarounds come to mind:
1. replace error by a warning. But for me, it defeats completely the purpose of the getters/setters
2. make sure the param/control always exists by giving them default values (the getters check the existence of the name in the list). But this will make the code extremely unsafe
3. Use tryCatch to convert error to warning. But again, unsafe.
4. Test the existence without the getters/setters. We'd loose in consistency but it will work

**Comment by andsv2** [comment]
Agreed; I imagine 2 and 4 are the most viable options however: for the infection prob. example I agree that that shouldn't occur and the error is the proper outcome there (will clean up the code to reflect that). For the second (and current example) something empty (like numeric( )) could work (and works with set and get) and would work with cond. logs. Not sure if this is behavior that is desired though (over NULL).

**Comment by smjenness** [comment]
5. (which I prefer): create an argument to these get functions, override.null.error, which is default FALSE, but TRUE in these cases. I think this will always apply to our built-in (toy) models only, and mostly in non-user facing/modified functions like the init modules. I would like to get a comprehensive list of all occurrences of this. @andsv2 - could you do that in the issue comment here?

**Comment by andsv2** [comment]
Will do.

**Comment by AdrienLeGuillou** [comment]
No problem for implementing something like that. 

So if I understand correctly, whent `override.null.error == TRUE` the functions returns `NULL` instead of an error if the name does not exist?

Another option would be to have `is_null_(attr/param/...)` that return TRUE if the element is not present OR equal to NULL

**Comment by smjenness** [comment]
Yes, the first option, please

**Comment by AdrienLeGuillou** [comment]
on it

**Comment by smjenness** [comment]
Closed with #381
